### PR TITLE
Add more test scenarios for data rehydration

### PR DESCRIPTION
**Pull request by lzhan**
Add more test scenarios for data rehydration

### CODE REVIEW

File: tests/func/applications/frameworkapp/common/application.json [hunk 1]
```diff
@@ -58,6 +58,9 @@
                     "LazyLoadRPC": {
                       "type": "LazyLoadRPC"
                     },
+                    "LazyLoadRPCCommon": {
+                      "type": "LazyLoadRPCCommon"
```
Comment (caridy) [inline tests/func/applications/frameworkapp/common/application.json]: I will like to remove RPC from that name, maybe `lazyLoadClientMojit`

File: tests/func/applications/frameworkapp/common/mojits/LazyLoadRPCCommon/binders/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                               + '<div id="LazyLoadRPCCommonbaz">'+ self.mojitProxy.data.get('bazc') + '</div>'
+                               + '<div id="LazyLoadRPCCommoncount">'+ 'Data has changed: '+count+' times</div>');
+                        }
+                });
+            }, this);
+        }
+
+    };
+
+}, '0.0.1', {requires: ['event-mouseenter', 'mojito-client']});
```
Comment (caridy) [inline tests/func/applications/frameworkapp/common/mojits/LazyLoadRPCCommon/binders/index.js]: `event-mouseenter` is not needed

File: tests/func/common/testlazyloadrpccommonclient.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+   
+    var suite = new Y.Test.Suite("Common: LazyLoadRPCClientCommon");
+
+    suite.add(new Y.Test.Case({
+
+        "test LazyLoadRPCCommonClient": function() {
+            var that = this;
+            Y.one('#lazyloadrpccommonButton').simulate('click');
+            that.wait(function(){
+                Y.log("here....");
```
Comment (caridy) [inline tests/func/common/testlazyloadrpccommonclient.js]: remove!

File: tests/func/common/testlazyloadrpccommonclient.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                Y.Assert.areEqual('Lazy load succeeded:', Y.one('#LazyLoadRPCCommontitle').get('innerHTML'));
+                Y.log("here...1.");
```
Comment (caridy) [inline tests/func/common/testlazyloadrpccommonclient.js]: remove!

**Comment by caridy** [comment]
@lzhan can you work on the notes that I left few days ago?

**Comment by lzhan** [comment]
@caridy I am working on it now. Jenny asked me to create a new card. So I moved this pr to the new card.

**Comment by caridy** [comment]
+1 (subject to travis result)

**Pull request merged**